lib/clAdapters/google/errors.ts

```diff
@@ -0,0 +1,40 @@
+export abstract class GoogleError extends Error {
```

Comment:
Ah good point. I'll give the union approach a shot and move some of the stack trace logic into the factory function. Stay tuned.